src/handlers/deeplinks.ts

```diff
@@ -326,3 +343,72 @@ function handleWalletConnect(uri?: string, connector?: string) {
     setHasPendingDeeplinkPendingRedirect(true);
   }
 }
+
+const querySwapAsset = async (uniqueId: string | undefined): Promise<ParsedSearchAsset | undefined> => {
+  if (!uniqueId) return undefined;
+
+  const { address, chainId } = getAddressAndChainIdFromUniqueId(uniqueId);
+  if (!supportedSwapChainIds.includes(parseInt(chainId.toString(), 10))) return undefined;
+  if (address !== 'eth' && address.length !== 42) return undefined;
+
+  const userAsset = userAssetsStore.getState().getUserAsset(uniqueId) || undefined;
+
+  const searchAsset = await queryTokenSearch({
+    chainId,
+    query: address.toLowerCase(),
+    keys: ['address'],
+    threshold: 'CASE_SENSITIVE_EQUAL',
+    list: 'verifiedAssets',
+  }).then(res => res[0]);
+
+  if (!searchAsset) return userAsset;
+
+  return parseSearchAsset({ searchAsset, userAsset });
+};
+
+function isValidGasSpeed(s: string | undefined): s is GasSpeed {
+  if (!s) return false;
+  return Object.values(GasSpeed).includes(s as GasSpeed);
+}
+
+async function setFromWallet(address: string | undefined) {
+  if (!address || !isAddress(address)) return;
+
+  const userWallets = store.getState().wallets.wallets!;
+  const wallet = Object.values(userWallets).find(w => w.addresses.some(a => a.address === address));
+
+  if (!wallet) return;
+
+  await Promise.all([store.dispatch(walletsSetSelected(wallet)), store.dispatch(addressSetSelected(address))]);
+}
+
+function isNumbericString(value: string | undefined): value is string {
+  if (!value) return false;
+  return !isNaN(+value);
+}
+
+async function handleSwapsDeeplink(url: string) {
+  const { query } = new URL(url, true);
+
+  await setFromWallet(query.from);
+
+  const params: SwapsParams = {};
+
+  const inputAsset = querySwapAsset(query.inputAsset);
+  const outputAsset = querySwapAsset(query.outputAsset);
+
+  if ('flashbots' in query) params.flashbots = query.flashbots === 'true';
+  if ('slippage' in query && isNumbericString(query.slippage)) params.slippage = query.slippage;
+
+  if (isNumbericString(query.percentageToSell)) params.percentageToSell = clamp(+query.percentageToSell, 0, 1);
+  else if (isNumbericString(query.inputAmount)) params.inputAmount = query.inputAmount;
+  else if (isNumbericString(query.outputAmount)) params.outputAmount = query.outputAmount;
```

Comment:
I don't want to make a style rule around this because it'd be annoying to enforce, but this is more difficult to read w/o brackets imo.